### PR TITLE
Feature/reinitialize

### PR DESCRIPTION
**Pull request by juergdrollinger**

### CODE REVIEW

File: src/main/java/ch/carve/microprofile/config/db/Repository.java [hunk 1]
```diff
@@ -38,36 +40,28 @@ public Repository(Config config) {
                 selectOne = datasource.getConnection().prepareStatement(queryOne);
                 selectAll = datasource.getConnection().prepareStatement(queryAll);
             } catch (SQLException e) {
-                log.debug("Configuration query could not be prepared: " + e.getMessage());
+                log.info("Configuration query could not be prepared: " + e.getMessage());
```
Comment (rikcarve) [inline src/main/java/ch/carve/microprofile/config/db/Repository.java:41]: ok for me, but then we need a way to disable the configurationsource

File: src/main/java/ch/carve/microprofile/config/db/Repository.java [hunk 2]
```diff
@@ -26,6 +26,8 @@
     PreparedStatement selectOne = null;
     PreparedStatement selectAll = null;
 
+    private Config config;
```
Comment (rikcarve) [inline src/main/java/ch/carve/microprofile/config/db/Repository.java]: is this needed?

File: src/main/java/ch/carve/microprofile/config/db/DatasourceConfigSource.java [hunk 3]
```diff
@@ -38,11 +46,17 @@ public String getValue(String propertyName) {
...
-            String value = repository.getConfigValue(propertyName);
-            cache.put(propertyName, new TimedEntry(value));
-            return value;
+            try {
+                String value = repository.getConfigValue(propertyName);
+                cache.put(propertyName, new TimedEntry(value));
+                return value;
+            } catch (SQLException e) {
+                log.info("query failed: " + e.getMessage());
+                clearRepository();
```
Comment (rikcarve) [inline src/main/java/ch/carve/microprofile/config/db/DatasourceConfigSource.java:55]: return null

File: src/main/java/ch/carve/microprofile/config/db/DatasourceConfigSource.java [hunk 4]
```diff
@@ -38,11 +46,17 @@ public String getValue(String propertyName) {
...
+            }
         }
-        return entry.getValue();
+        // if the entry was never cached, then it will be null
+        return entry != null ? entry.getValue() : null;
```
Comment (rikcarve) [inline src/main/java/ch/carve/microprofile/config/db/DatasourceConfigSource.java:59]: then leave this as is

**Review by rikcarve — CHANGES_REQUESTED** [review]

File: src/main/java/ch/carve/microprofile/config/db/DatasourceConfigSource.java (hunk 4 above)
Comment (juergdrollinger) [reply src/main/java/ch/carve/microprofile/config/db/DatasourceConfigSource.java:59]: yeaph, thanks for pointing to a cleaner solution

File: src/main/java/ch/carve/microprofile/config/db/Repository.java (hunk 2 above)
Comment (juergdrollinger) [reply src/main/java/ch/carve/microprofile/config/db/Repository.java]: no

File: src/main/java/ch/carve/microprofile/config/db/DatasourceConfigSource.java (hunk 4 above)
Comment (juergdrollinger) [reply src/main/java/ch/carve/microprofile/config/db/DatasourceConfigSource.java:59]: as just found out: If the entry is expired and the sql fails, then reuse the expired value.

**Review by rikcarve — APPROVED** [review]

**Pull request merged**